src/components/AutoCompleteInput/AutoCompleteInput.js

```diff
@@ -116,21 +173,44 @@ export default class AutoCompleteInput extends Component {
     if (value) {
       onChange(value);
 
-      if (clearOnSelect && this.downshiftRef) {
-        this.downshiftRef.clearSelection();
+      if (clearOnSelect) {
+        this.handleClear();
       }
     }
   };
 
+  handleClear = () => {
+    if (this.downshiftRef) {
+      this.downshiftRef.clearSelection();
+    }
+  };
+
   handleDownShiftRef = ref => {
     this.downshiftRef = ref;
   };
 
   render() {
-    const { items, onChange, clearOnSelect, ...inputProps } = this.props;
+    const {
+      items,
```

Comment:
I want to check my understanding here-- `items` is not used below and is re-mapped to `options`, but is passed here because it was the previous implementation and we need to give a warning that it's being deprecated, is that right?

Reply:
That's correct. We can't remove the prop because that would be a breaking change. Instead, we mark it as deprecated to give users a head's up that they need to adjust their code. The prop will be completely removed in the next major version (the upcoming v2).